ui/dmt_yieldcalculator.lua

```diff
@@ -81,19 +87,41 @@ function UpdatePinYields(playerID:number, pinsToUpdate:table)
             pin.CanPlace = canPlace;
             pin.CanPlaceToolTip = canPlaceToolTip;
             -- Update the pin in MapPinSubjectManager.
-            MapPinSubjectManager.UpdatePin(playerID, pin.X, pin.Y, pin);
+            UpdateMapPinSubject(playerID, pin.X, pin.Y, pin);
         else
-            MapPinSubjectManager.ClearPin(playerID, pin.X, pin.Y);
+            ClearMapPinSubject(playerID, pin.X, pin.Y);
         end
         -- Update UI.
         LuaEvents.DMT_RefreshMapPinUI(playerID, pin.Id);
     end
+
+    -- Send the MapPin subjects to the other clients
+    if (playerID ~= -1 and playerID == Game.GetLocalPlayer()) then
+        -- Attempt to broadcast to the other clients
+        local bSuccess = BroadcastMapPinSubjects(playerID);
+
+        -- Update the cache based on the result of the broadcast attempt
+        m_CachedMapPinSubjects = bSuccess and {} or GetAllMapPinSubjects(playerID);
+
+        -- Get the current state of the restore map pin subjects config
+        local playerConfig = PlayerConfigurations[playerID];
+        local restoreSubjectsConfig = playerConfig:GetValue(RESTORE_MAP_PIN_SUBJECTS_CONFIG_KEY);
+
+        -- Update the config if the state should be changed
+        if (not bSuccess and restoreSubjectsConfig ~= 1) then
+            playerConfig:SetValue(RESTORE_MAP_PIN_SUBJECTS_CONFIG_KEY, 1);
+            Network.BroadcastPlayerInfo();
```

Comment:
I'm glad you brought that up, because I neglected to mention that in my original post. When I originally was researching how to go about this, I thought about doing that initially. Like you suspected though, you can't store tables in the player configuration. I decided to go with the solution I originally had here because I was worried about the potential performance impact of the serialization / deserialization functions.

After reading your comment though, I reconsidered the idea and I agree with you. The solution I originally had here became more complex than I originally thought it would, and ended up having some issues come up later on. It mostly worked, but had some issues occurring seemingly randomly (probably due to how Civ 6 handles networking?). These issues go away when saving to the player configuration instead though. In addition, the player configuration solution is much simpler since it doesn't need the cache/restore logic, like you pointed out.

So, after reconsidering, I replaced my original solution here with the player configuration idea. I haven't tested it thoroughly yet, so I'll leave it as a draft until I do in the next couple days. I have tested the performance a little bit though. The serialization / deserialization functions I used seem to make no noticeable impact on performance.

The `Network.BroadcastPlayerInfo()` function seems to have a very minor impact on performance in multiplayer games, but only by a small fraction of a second at most when placing/editing pins. The impact it has varies and sometimes is negligible. To keep it as minimal as possible though, I only call it at the end of the `UpdatePinYields()` function. Due to how the base game MapPinManager is written, calling this function also refreshes all the pins on the map when it is called. As such, I was able to remove the `RefreshMapPinUI` LuaEvent, and everything still updates as expected.

Sorry for the wall of text, but I wanted to explain all my thoughts there. I look forward to hearing your thoughts if you have any!